src/QsCompiler/DataStructures/ReservedKeywords.fs

```diff
@@ -198,6 +198,9 @@ module InternalUse =
     /// to be used as name for all namespaces that do not have a valid name
     let UnknownNamespace = "__UnknownNamespaceName__"
 
+    /// the implicit namespace in notebooks
+    let NotebookNamespace = "__NotebookNamespace__"
```

Comment:
nit: I kind of prefer just `__Notebook__` and `__Unknown__`, since by context we know they are namespaces.